src/unstract/adapters/llm/llm_adapter.py

```diff
@@ -48,3 +49,16 @@ def get_llm_instance(self) -> Optional[LLM]:
 
     def test_connection(self, llm_metadata: dict[str, Any]) -> bool:
         return False
+
+    def get_context_window_size(self) -> int:
+        """Get the context window size supported by the LLM.
+
+        Returns:
+            int: Context window size supported by the LLM
+        """
+        context_window_size: int = 0
+        llm = self.get_llm_instance()
+        if llm is not None:
+            context_window_size = llm.metadata.context_window
```

Comment:
What is the default context_window if we are not set, Is it None?
Is the llm.metadata always conatins context_window ?

Reply:
context_window is not stored anywhere for now. It is dynamically fetched only while constructing the response for the Get adapter details API.

Reply:
I just added a constant to avoid a magic number for readability.

Reply:
@gaya3-zipstack why do we return `Optional[LLM]` in `get_llm_instance()`? This leads to an extra check at all places of use - and in most scenarios we raise an exception. Instead can we always raise an exception if no valid LLM is found? Additionally we can control the behavior of raising an exception with a flag

Reply:
@chandrasekharan-zipstack yeah we could do that. That will be a good improvement. Needs to be done in the SDK. Can we decouple it with this PR?